File: super-impactor source/C#/DownloadFiles.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics;
using System.Drawing;
using System.IO;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace SuperImpactor
{
	[DesignerGenerated]
	public class DownloadFiles : Form
	{
		private IContainer components;

		internal virtual Button Button1
		{
			get
			{
				stackVariable1 = this._Button1;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.Button1_Click);
				Button button = this._Button1;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._Button1 = value;
				button = this._Button1;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Button cmdClose
		{
			get
			{
				stackVariable1 = this._cmdClose;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdClose_Click);
				Button button = this._cmdClose;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdClose = value;
				button = this._cmdClose;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Button cmdRemove
		{
			get
			{
				stackVariable1 = this._cmdRemove;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdRemove_Click);
				Button button = this._cmdRemove;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdRemove = value;
				button = this._cmdRemove;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Button cmdRemoveAll
		{
			get
			{
				stackVariable1 = this._cmdRemoveAll;
				return stackVariable1;
			}
			[MethodImpl(MethodImplOptions.Synchronized)]
			set
			{
				EventHandler eventHandler = new EventHandler(this.cmdRemoveAll_Click);
				Button button = this._cmdRemoveAll;
				if (button != null)
				{
					button.Click -= eventHandler;
				}
				this._cmdRemoveAll = value;
				button = this._cmdRemoveAll;
				if (button != null)
				{
					button.Click += eventHandler;
				}
			}
		}

		internal virtual Label Label1
		{
			get;
			[MethodImpl(MethodImplOptions.Synchronized)]
			set;
		}

		internal virtual ListView lstCydia
		{
			get;
			[MethodImpl(MethodImplOptions.Synchronized)]
			set;
		}

		public DownloadFiles()
		{
			base.Load += new EventHandler(this.DownloadFiles_Load);
			this.InitializeComponent();
		}

		private void Button1_Click(object sender, EventArgs e)
		{
			if (this.lstCydia.SelectedItems.Count != 0)
			{
				ListViewItem item = this.lstCydia.SelectedItems[0];
				Install install = new Install();
				install.installFromFile(string.Concat(AppConst.m_rootPath, "\\download\\", item.SubItems[0].Text), "", "");
			}
			else
			{
				Interaction.MsgBox("Please select file to install", MsgBoxStyle.OkOnly, null);
			}
		}

		private void cmdClose_Click(object sender, EventArgs e)
		{
			base.Close();
		}

		private void cmdRemove_Click(object sender, EventArgs e)
		{
			IEnumerator enumerator = null;
			try
			{
				enumerator = this.lstCydia.SelectedItems.GetEnumerator();
				while (enumerator.MoveNext())
				{
					ListViewItem current = (ListViewItem)enumerator.Current;
					File.Delete(string.Concat(AppConst.m_rootPath, "\\download\\", current.SubItems[0].Text));
					current.Remove();
				}
			}
			finally
			{
				if (enumerator is IDisposable)
				{
					(enumerator as IDisposable).Dispose();
				}
			}
		}

		private void cmdRemoveAll_Click(object sender, EventArgs e)
		{
			Common.DeleteFilesFromFolder(string.Concat(AppConst.m_rootPath, "\\download\\"));
			this.lstCydia.Clear();
		}

		[DebuggerNonUserCode]
		protected override void Dispose(bool disposing)
		{
			try
			{
				if ((!disposing ? false : this.components != null))
				{
					this.components.Dispose();
				}
			}
			finally
			{
				base.Dispose(disposing);
			}
		}

		private void DownloadFiles_Load(object sender, EventArgs e)
		{
			string str;
			double num;
			this.lstCydia.Clear();
			this.lstCydia.View = View.Details;
			this.lstCydia.Columns.Add("Name", checked(this.lstCydia.Width - 70), HorizontalAlignment.Left);
			this.lstCydia.Columns.Add("Size");
			string[] files = Directory.GetFiles(string.Concat(AppConst.m_rootPath, "\\download\\"));
			int num1 = Information.UBound(files, 1);
			for (int i = 0; i <= num1; i = checked(i + 1))
			{
				ListViewItem listViewItem = new ListViewItem();
				FileInfo fileInfo = new FileInfo(files[i]);
				listViewItem.Text = fileInfo.Name;
				if ((double)fileInfo.Length / 1024 / 1024 / 1024 >= 1)
				{
					num = Math.Round((double)fileInfo.Length / 1024 / 1024 / 1024, 2);
					str = string.Concat(num.ToString(), " GB");
				}
				else if ((double)fileInfo.Length / 1024 / 1024 < 1)
				{
					num = Math.Round((double)fileInfo.Length / 1024, 2);
					str = string.Concat(num.ToString(), " KB");
				}
				else
				{
					num = Math.Round((double)fileInfo.Length / 1024 / 1024, 2);
					str = string.Concat(num.ToString(), " MB");
				}
				listViewItem.SubItems.Add(str);
				this.lstCydia.Items.Add(listViewItem);
			}
		}

		[DebuggerStepThrough]
		private void InitializeComponent()
		{
			this.cmdClose = new Button();
			this.cmdRemove = new Button();
			this.cmdRemoveAll = new Button();
			this.Label1 = new Label();
			this.lstCydia = new ListView();
			this.Button1 = new Button();
			base.SuspendLayout();
			this.cmdClose.Location = new Point(432, 361);
			this.cmdClose.Name = "cmdClose";
			this.cmdClose.Size = new System.Drawing.Size(104, 25);
			this.cmdClose.TabIndex = 9;
			this.cmdClose.Text = "Close";
			this.cmdClose.UseVisualStyleBackColor = true;
			this.cmdRemove.Location = new Point(432, 61);
			this.cmdRemove.Name = "cmdRemove";
			this.cmdRemove.Size = new System.Drawing.Size(104, 25);
			this.cmdRemove.TabIndex = 8;
			this.cmdRemove.Text = "Remove";
			this.cmdRemove.UseVisualStyleBackColor = true;
			this.cmdRemoveAll.Location = new Point(432, 94);
			this.cmdRemoveAll.Name = "cmdRemoveAll";
			this.cmdRemoveAll.Size = new System.Drawing.Size(104, 25);
			this.cmdRemoveAll.TabIndex = 7;
			this.cmdRemoveAll.Text = "Remove All";
			this.cmdRemoveAll.UseVisualStyleBackColor = true;
			this.Label1.AutoSize = true;
			this.Label1.Location = new Point(12, 9);
			this.Label1.Name = "Label1";
			this.Label1.Size = new System.Drawing.Size(110, 13);
			this.Label1.TabIndex = 6;
			this.Label1.Text = "List of Download Files";
			this.lstCydia.FullRowSelect = true;
			this.lstCydia.GridLines = true;
			this.lstCydia.HideSelection = false;
			this.lstCydia.Location = new Point(12, 28);
			this.lstCydia.Name = "lstCydia";
			this.lstCydia.Size = new System.Drawing.Size(405, 358);
			this.lstCydia.TabIndex = 5;
			this.lstCydia.UseCompatibleStateImageBehavior = false;
			this.Button1.Location = new Point(432, 28);
			this.Button1.Name = "Button1";
			this.Button1.Size = new System.Drawing.Size(104, 25);
			this.Button1.TabIndex = 10;
			this.Button1.Text = "Install";
			this.Button1.UseVisualStyleBackColor = true;
			base.AutoScaleDimensions = new SizeF(6f, 13f);
			base.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
			base.ClientSize = new System.Drawing.Size(548, 398);
			base.ControlBox = false;
			base.Controls.Add(this.Button1);
			base.Controls.Add(this.cmdClose);
			base.Controls.Add(this.cmdRemove);
			base.Controls.Add(this.cmdRemoveAll);
			base.Controls.Add(this.Label1);
			base.Controls.Add(this.lstCydia);
			base.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
			base.MaximizeBox = false;
			base.MinimizeBox = false;
			base.Name = "DownloadFiles";
			this.Text = "DownloadFiles";
			base.ResumeLayout(false);
			base.PerformLayout();
		}
	}
}

File: VB/DownloadFiles.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.IO
Imports System.Runtime.CompilerServices
Imports System.Windows.Forms

Namespace SuperImpactor
	<DesignerGenerated>
	Public Class DownloadFiles
		Inherits Form
		Private components As IContainer

		Friend Overridable Property Button1 As Button
			Get
				stackVariable1 = Me._Button1
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.Button1_Click)
				Dim button As System.Windows.Forms.Button = Me._Button1
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._Button1 = value
				button = Me._Button1
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property cmdClose As Button
			Get
				stackVariable1 = Me._cmdClose
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdClose_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdClose
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdClose = value
				button = Me._cmdClose
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property cmdRemove As Button
			Get
				stackVariable1 = Me._cmdRemove
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdRemove_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdRemove
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdRemove = value
				button = Me._cmdRemove
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property cmdRemoveAll As Button
			Get
				stackVariable1 = Me._cmdRemoveAll
				Return stackVariable1
			End Get
			<MethodImpl(MethodImplOptions.Synchronized)>
			Set(ByVal value As System.Windows.Forms.Button)
				Dim eventHandler As System.EventHandler = New System.EventHandler(AddressOf Me.cmdRemoveAll_Click)
				Dim button As System.Windows.Forms.Button = Me._cmdRemoveAll
				If (button IsNot Nothing) Then
					RemoveHandler button.Click,  eventHandler
				End If
				Me._cmdRemoveAll = value
				button = Me._cmdRemoveAll
				If (button IsNot Nothing) Then
					AddHandler button.Click,  eventHandler
				End If
			End Set
		End Property

		Friend Overridable Property Label1 As Label

		Friend Overridable Property lstCydia As ListView

		Public Sub New()
			MyBase.New()
			AddHandler MyBase.Load,  New EventHandler(AddressOf Me.DownloadFiles_Load)
			Me.InitializeComponent()
		End Sub

		Private Sub Button1_Click(ByVal sender As Object, ByVal e As EventArgs)
			If (Me.lstCydia.SelectedItems.Count <> 0) Then
				Dim item As ListViewItem = Me.lstCydia.SelectedItems(0)
				Dim install As SuperImpactor.Install = New SuperImpactor.Install()
				install.installFromFile(String.Concat(AppConst.m_rootPath, "\download\", item.SubItems(0).Text), "", "")
			Else
				Interaction.MsgBox("Please select file to install", MsgBoxStyle.OkOnly, Nothing)
			End If
		End Sub

		Private Sub cmdClose_Click(ByVal sender As Object, ByVal e As EventArgs)
			MyBase.Close()
		End Sub

		Private Sub cmdRemove_Click(ByVal sender As Object, ByVal e As EventArgs)
			Dim enumerator As IEnumerator = Nothing
			Try
				enumerator = Me.lstCydia.SelectedItems.GetEnumerator()
				While enumerator.MoveNext()
					Dim current As ListViewItem = DirectCast(enumerator.Current, ListViewItem)
					File.Delete(String.Concat(AppConst.m_rootPath, "\download\", current.SubItems(0).Text))
					current.Remove()
				End While
			Finally
				If (TypeOf enumerator Is IDisposable) Then
					TryCast(enumerator, IDisposable).Dispose()
				End If
			End Try
		End Sub

		Private Sub cmdRemoveAll_Click(ByVal sender As Object, ByVal e As EventArgs)
			Common.DeleteFilesFromFolder(String.Concat(AppConst.m_rootPath, "\download\"))
			Me.lstCydia.Clear()
		End Sub

		<DebuggerNonUserCode>
		Protected Overrides Sub Dispose(ByVal disposing As Boolean)
			Try
				If (If(Not disposing, False, Me.components IsNot Nothing)) Then
					Me.components.Dispose()
				End If
			Finally
				MyBase.Dispose(disposing)
			End Try
		End Sub

		Private Sub DownloadFiles_Load(ByVal sender As Object, ByVal e As EventArgs)
			Dim str As String
			Dim num As Double
			Me.lstCydia.Clear()
			Me.lstCydia.View = View.Details
			Me.lstCydia.Columns.Add("Name", Me.lstCydia.Width - 70, HorizontalAlignment.Left)
			Me.lstCydia.Columns.Add("Size")
			Dim files As String() = Directory.GetFiles(String.Concat(AppConst.m_rootPath, "\download\"))
			Dim num1 As Integer = Information.UBound(files, 1)
			For i As Integer = 0 To num1 Step 1
				Dim listViewItem As System.Windows.Forms.ListViewItem = New System.Windows.Forms.ListViewItem()
				Dim fileInfo As System.IO.FileInfo = New System.IO.FileInfo(files(i))
				listViewItem.Text = fileInfo.Name
				If (CDbl(fileInfo.Length) / 1024 / 1024 / 1024 >= 1) Then
					num = Math.Round(CDbl(fileInfo.Length) / 1024 / 1024 / 1024, 2)
					str = String.Concat(num.ToString(), " GB")
				ElseIf (CDbl(fileInfo.Length) / 1024 / 1024 < 1) Then
					num = Math.Round(CDbl(fileInfo.Length) / 1024, 2)
					str = String.Concat(num.ToString(), " KB")
				Else
					num = Math.Round(CDbl(fileInfo.Length) / 1024 / 1024, 2)
					str = String.Concat(num.ToString(), " MB")
				End If
				listViewItem.SubItems.Add(str)
				Me.lstCydia.Items.Add(listViewItem)
			Next

		End Sub

		<DebuggerStepThrough>
		Private Sub InitializeComponent()
			Me.cmdClose = New Button()
			Me.cmdRemove = New Button()
			Me.cmdRemoveAll = New Button()
			Me.Label1 = New Label()
			Me.lstCydia = New ListView()
			Me.Button1 = New Button()
			MyBase.SuspendLayout()
			Me.cmdClose.Location = New Point(432, 361)
			Me.cmdClose.Name = "cmdClose"
			Me.cmdClose.Size = New System.Drawing.Size(104, 25)
			Me.cmdClose.TabIndex = 9
			Me.cmdClose.Text = "Close"
			Me.cmdClose.UseVisualStyleBackColor = True
			Me.cmdRemove.Location = New Point(432, 61)
			Me.cmdRemove.Name = "cmdRemove"
			Me.cmdRemove.Size = New System.Drawing.Size(104, 25)
			Me.cmdRemove.TabIndex = 8
			Me.cmdRemove.Text = "Remove"
			Me.cmdRemove.UseVisualStyleBackColor = True
			Me.cmdRemoveAll.Location = New Point(432, 94)
			Me.cmdRemoveAll.Name = "cmdRemoveAll"
			Me.cmdRemoveAll.Size = New System.Drawing.Size(104, 25)
			Me.cmdRemoveAll.TabIndex = 7
			Me.cmdRemoveAll.Text = "Remove All"
			Me.cmdRemoveAll.UseVisualStyleBackColor = True
			Me.Label1.AutoSize = True
			Me.Label1.Location = New Point(12, 9)
			Me.Label1.Name = "Label1"
			Me.Label1.Size = New System.Drawing.Size(110, 13)
			Me.Label1.TabIndex = 6
			Me.Label1.Text = "List of Download Files"
			Me.lstCydia.FullRowSelect = True
			Me.lstCydia.GridLines = True
			Me.lstCydia.HideSelection = False
			Me.lstCydia.Location = New Point(12, 28)
			Me.lstCydia.Name = "lstCydia"
			Me.lstCydia.Size = New System.Drawing.Size(405, 358)
			Me.lstCydia.TabIndex = 5
			Me.lstCydia.UseCompatibleStateImageBehavior = False
			Me.Button1.Location = New Point(432, 28)
			Me.Button1.Name = "Button1"
			Me.Button1.Size = New System.Drawing.Size(104, 25)
			Me.Button1.TabIndex = 10
			Me.Button1.Text = "Install"
			Me.Button1.UseVisualStyleBackColor = True
			MyBase.AutoScaleDimensions = New SizeF(6!, 13!)
			MyBase.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font
			MyBase.ClientSize = New System.Drawing.Size(548, 398)
			MyBase.ControlBox = False
			MyBase.Controls.Add(Me.Button1)
			MyBase.Controls.Add(Me.cmdClose)
			MyBase.Controls.Add(Me.cmdRemove)
			MyBase.Controls.Add(Me.cmdRemoveAll)
			MyBase.Controls.Add(Me.Label1)
			MyBase.Controls.Add(Me.lstCydia)
			MyBase.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
			MyBase.MaximizeBox = False
			MyBase.MinimizeBox = False
			MyBase.Name = "DownloadFiles"
			Me.Text = "DownloadFiles"
			MyBase.ResumeLayout(False)
			MyBase.PerformLayout()
		End Sub
	End Class
End Namespace